### ISSUE TITLE
User not able to navigate after making changes to a field in balance page two times and other things happen

### ISSUE THREAD

**Issue by dot-william** [wappints/CSSWENGFinal-Group3#9]
### Description
- User is unable to navigate to another page after making changes to a field in balance page two times or more and then additional sale entries are added to the home page when the app is restarted

### Steps to Reproduce:
1. Go to balance page
2. Input in payment field at any entry the value '1'
3. Click enter
4. Click enter again 
5. Click x amount of times

### Bonus bug
1. Close the program
2. Go to home page, you will see duplicate entries of the balance edited (see screenshot 1)

### Expected Results:
- A total of 2 was subtracted to the balance
- Change reflected to the db
- User will be able to navigate to another page

### Actual Results:
-  A total of 2 was subtracted to the balance 
- Change **not reflected** to the db
- User is **unable** to navigate to another page, rendering the app unusable
- It broke the system - I suddenly see new duplicated entries in the home page

### Screenshots 
![image](https://user-images.githubusercontent.com/60463265/148647210-26687f06-1700-441b-915b-956e1de64eea.png)

**Comment by wappints** [comment]
Bonus Bug <-- not a bug, assume that whenever user enters a value to the payment input, then no matter how much they press enter with the same value, it adds an entry with that given value, and the balance should be expected to be negative which signifies that it is a payment to a certain balance / debt. On the side note, mine reflected in the DB. 

I'll just add validation when pressing enter (input field will be empty upon trigger event)

**Comment by wappints** [comment]
Tried everything. It works so far. Just remember that whenever you press enter, it will automatically add an entry in the HOME as if it were a "partial payment" or "installment" to the parent balance. 

I did a lot of trial and error to do the navigation problem, but it did not occur to me. It is most likely going to occur if you keep tampering with every values and buttons too fast. This is just a guess though based on previous experience.

Fixed.
